### DISCUSSION TITLE
Why is `shouldCloseOnInteractOutside` prevented if the event had preventDefault?

### DISCUSSION THREAD

**Discussion by AndreasHald** [melt-ui/melt-ui#1241] [General]
see code here https://github.com/melt-ui/melt-ui/blob/23129c189c224b69adec947366a7120722982068/src/lib/builders/popover/create.ts#L152

 were using svelte kit shallow routing (https://kit.svelte.dev/docs/shallow-routing#loading-data-for-a-route) to manage modals which entails having an anchor that links to somewhere else, and on click prevent default and use `pushState(href, { selected: result.data });` to open a modal instead.
 
 However clicking this link while having a melt dropdown / popover open fails to close the popover. because of the mentioned code.
 
 What is the reason for this check?

**Comment by TGlide** [comment]
The only thing that this line *should* do, is that it won't close the popover if the "click outside" custom event was prevented, not any click event. Not saying its what is currently happening, but its the purpose behind it, similar to the check we have in all event listeners.

Could you share a repro with the issue you're facing, and what's the expected outcome?

**Reply by AndreasHald** [reply]
Ofcourse repo created [here](https://stackblitz.com/edit/github-aoqh8h?file=src%2Froutes%2F%2Bpage.svelte) 

When clicking the popover button the popover opens as expected, any click outside of the popover will close the popover as expected, except If the link that does shallow navigation is clicked. if that link is clicked the popover remains open.

I'm not even sure if it's a bug or intended behaviour because the click handler for the link does prevent default on the click event, but it stumped me because I think I would except it to close the popover

**Reply by TGlide** [reply]
I see! Thanks for the repro. Will convert this to an issue, definitely an issue on our end

**Reply by AndreasHald** [reply]
Cool :) I know you're busy. So if it's easier for you and you can explain a bit more about the custom event so I don't break that logic, I can try creating a PR for it.

**Reply by TGlide** [reply]
It's a bit complex: We have an `interact-outside` action, and currently, it just gives you the original event, and we run defaultPrevented checks on them (wrongly). 

The quick fix is to not listen for default prevents at all.
The correct fix is to instead ship a custom event, that can be prevented (while not preventing the original trigger event), and that also ships the trigger event so we can extract info.
